util/codec/bytes.go

```diff
@@ -47,9 +47,12 @@ import (
 // EncodeBytes guarantees the encoded value is in ascending order for comparison,
 // The encoded value is >= SmallestNoneNilValue and < InfiniteValue.
 func EncodeBytes(b []byte, data []byte) []byte {
+	// Allocate more space to avoid unnecessary slice growing
+	bs := make([]byte, len(b), len(b)+len(data)+20)
```

Comment:
We write code rarely like that.  We just call EncodeBytes with nil as argument in most time.